modules/systemtests/src-test/org/ejbca/core/ejb/ca/caadmin/RenewCATest.java

```diff
@@ -97,6 +99,84 @@ public void test01renewCA() throws Exception {
+            .getRemoteSession(CryptoTokenManagementSessionRemote.class);
+    private final CertificateProfileSessionRemote certificateProfileSession = EjbRemoteHelper.INSTANCE
+            .getRemoteSession(CertificateProfileSessionRemote.class);
+
+    @Test
+    public void test02renewCA_ChangeKeyAlg() throws Exception {
+        log.trace(">test02renewCA_ChangeKeyAlg()");
+        
+        X509CAInfo info = (X509CAInfo) caSession.getCAInfo(internalAdmin, "TEST");
+        X509Certificate orgcert = (X509Certificate) info.getCertificateChain().iterator().next();
+        // Sleep at least for one second so we are not so fast that we create a new cert with the same time
+        Thread.sleep(2000);
+        
+        // Prepare to renew CA but with an EC key
+        
+        // Get the CA's token
+        final CAToken caToken = info.getCAToken();
+        
+        // The current Signing Algorithm should be RSA-based
+        String sPreviousSigAlg = ((X509Certificate)orgcert).getSigAlgName();
+        assertTrue("Current CA's Signature Algorithm should include RSA", sPreviousSigAlg.contains("RSA"));
+
+        // Set the next key alias for the token
+        String sNextKeyAlias = "TestEC";
+        
+        // Create an EC key. Need the CryptoTokeManagementSession for this
+        cryptoTokenManagementSession.createKeyPair(internalAdmin, caToken.getCryptoTokenId(), sNextKeyAlias, com.keyfactor.util.keys.token.KeyGenParams.builder("prime256v1").build());
+
+        // To get EJBCA to renew a CA with a different key algorithm, we need to:
+        //   1. set up the signing algorithm in the CA's token to support EC, 
+        //   2. ensure the certificate profile has an appropriate signature algorithm to support EC.
+        
+        // Set the signature algorithm of the token
+        caToken.setSignatureAlgorithm( AlgorithmConstants.SIGALG_SHA256_WITH_ECDSA);
+        
+        // Need to set the Certificate Profile to use ECDSA based signature algorithm
+        // Lets copy the current profile, change it, and save a new profile (as we can't edit the current profile)
+        int iCP = info.getCertificateProfileId();
+        org.cesecore.certificates.certificateprofile.CertificateProfile cpCA = certificateProfileSession.getCertificateProfile(iCP);
+        cpCA.setSignatureAlgorithm( AlgorithmConstants.SIGALG_SHA256_WITH_ECDSA);
+        iCP = certificateProfileSession.addCertificateProfile(internalAdmin, "TESTRENEWALWITHEC", cpCA);
+        // Update the CA
+        info.setCertificateProfileId(iCP);
+        caSession.editCA(internalAdmin, info);
+
+        // We are all set and now ready to renew the CA
+        caAdminSession.renewCA(internalAdmin, info.getCAId(), sNextKeyAlias, null, /*CreateLinkCert*/true);
+
+        // Remove the certificate profile we just created.
+        certificateProfileSession.removeCertificateProfile(internalAdmin, "TESTRENEWALWITHEC");
```

Comment:
use try/finally to remove things that should be removed after the test. It ensures cleanup is done in case of test failures.

Reply:
Yep, can do.

Reply:
Also, this is a good idea for the main code to ensure any SignatureAlgorithm settings are put back if an error occurs creating the Link cert. So I have encluded this into the CAAdminSessionBean.